### PR TITLE
release: pre-main → main (13 PRs since 1.86.0) - the Windows port, and a walkthrough that no longer stalls

### PR DESCRIPTION
**Pull request by Akarsh-Hegde**
Promotes `pre-main` to production. **13 PRs since 1.86.0** (both branches currently sit on the `chore(release): 1.86.0` commit, so this window is everything merged after it). `feat:` commits are present, so semantic-release will cut **1.87.0**.

Theme of the window: **the Windows port stops being second-class, and the first-run walkthrough stops stalling.** Nine of the thirteen are one or the other.

> ### ⚠️ Read the diff with this in mind
> `main` and `pre-main` have **two merge bases** (`498c3b80` and `23f24797`), and GitHub has picked the older one. So the diff above shows **52 files**, not the real **46** (+3619 / -326), and the extra six are an artifact:
>
> - `Cargo.toml`, `tray/src-tauri/Cargo.toml`, `ui/package.json`, `packages/meridian-mcp/package.json` - each shows `1.85.0 → 1.86.0`. **Both branches are already on 1.86.0.** This bump is not in this release.
> - `CHANGELOG.md` - shows the 1.86.0 section being added. **It already exists on `main`.**
>
> Any review comment on those five files is reviewing something that is already merged. The real content of this PR is the 43 files under `tray/`, `ui/`, `docs/` and `.github/`.

---

## Windows

Four defects, all of them things macOS structurally cannot catch - they are `cfg(target_os = "windows")` code or Win32 window-state semantics, so a green macOS run says nothing about any of them. Windows CI is their first and only compile.

### #764 - the dashboard was a trap
`sys::open_full_screen` called `set_fullscreen(true)` with no platform gate. On macOS that is native full-screen and comes with its own exit. On Windows tao maps it to `Fullscreen::Borderless`, which strips the title bar, sizes to the **full monitor rect** rather than the work area, and calls `taskbar_mark_fullscreen`. No title bar, no taskbar, and nothing in the app binds F11 or Escape on that window - the only ways out were Alt+F4 and Alt+Tab. Now gated to macOS; Windows keeps `fill_work_area`, which was already the correct answer there.

### #768 - the setup wizard un-maximized itself
Two call sites resize the setup window between steps (`resize_setup_window`, and `nudgeWindowRepaint`'s WKWebView compositor workaround). Both guarded only against native full-screen, and on Windows `is_fullscreen()` is `false` for a *maximized* window. Win32 has no "resize while staying maximized" - `SetWindowPos` on a `WS_MAXIMIZE` window clears the state - so every Continue press visibly restored the window down. Both now skip a maximized window, and the nudge checks the **pair** (`maximized && !fullscreen`) because a macOS window in native full-screen also reports `isMaximized()` true. Same PR: the footer CTA moves to the product accent `#7C3AED` (white on it clears 4.5:1; the old fill cleared neither label colour), and the Windows-only Alerts step is skipped when it opens already `GRANTED`, which is the common case.

### #771 - the installer and the watchdog fought over the daemon
An install staged a new daemon binary while the health watchdog was still running, so the watchdog could respawn the daemon mid-copy. Closed with a staging flag held by an RAII guard across stop -> copy -> register. The TOCTOU is genuinely closed rather than narrowed: the watchdog needs two consecutive strikes, and the daemon is only stopped after the flag is set. Follow-up in the same window (`20656afa`) kills by the pid list the probe just enumerated instead of re-running a full `Get-CimInstance` up to 40 more times per teardown, and adds the missing test for the health-poll gate.

### #776 - interactive toasts actually work
Windows toasts were plain title+body. An interactive notification delivered but could **not be answered** - its outbox row was only ever resolved by expiry - and `retract_toast` always failed, leaving a toast in Action Center with live buttons. Neither was a Windows limitation: it was the plugin's `notify-rust` backend, whose `set_click_listener_active` and `remove_active` are both hard `Err`. The outbox path now drives `Windows.UI.Notifications` directly (`win_toast.rs`, `toast_actions.rs`), and answers land on the outbox row through the same `apply_response` the macOS plugin path uses, so both platforms write identically. Also removes a spurious `notification category registration failed` ERROR on every Windows launch - categories are a `UNUserNotificationCenter` concept and are now a documented no-op there.

**Reviewer note on #776:** plain `sys::notify` toasts deliberately stay on the plugin, so Windows runs two toast stacks under one AUMID; everything runs on the main thread because WinRT needs an initialised COM apartment and the tokio poll workers have neither; and there is intentionally **no COM activator**, which means answers after a tray restart are dropped - the same constraint macOS already has.

---

### #770 - the tour types the first task
The planner's opening beat rang an empty box and waited up to three minutes for the user to invent a task, *before* they had seen Meridian draft anything. The beat straight after - pressing **Draft with AI** - is the one that shows the product doing something, and a user parked at the blank box never reached it. The tour now types `setting up meridian and going through onboarding`, which lands in the user's **real** plan and is therefore both a good demonstration and true. Typed through the prototype's native value setter with a bubbling `input` event, because `el.value = x` updates neither React's tracker nor its state - the text would show for one frame and then be wiped, silently.

### #769 - Generate worklog asked for a model at the right moment
Two failures compounded. The "Choose a provider" button opened Settings **behind** the dialog the user was looking at (`SettingsModal` is `z-40` inside `ModalShell`; `WorklogDraftDialog` is `z-50` portalled to `document.body`, and the shell's root opens no stacking context), so the one path the product offers for connecting a model appeared to do nothing. And the walkthrough bypassed the check entirely (`if (isDemo) { generate(); return }`), demonstrating a finished worklog on a machine with no AI configured - teaching a flow that cannot happen, to the audience least able to tell.

### #767 - the spotlight's corners
The dimmed beat drew a rounded ring around a rounded card and cut a **square** hole in the dim behind it. The existing comment claimed the ring hid it; the numbers said otherwise (ring at 6px offset, hole at 8px - the hole was the larger, so it was never underneath). The four panes stay, since the pointer fence depends on them - only the dim moves onto a scrim with a `box-shadow` spread, which is the one way to get a rounded hole without a mask WebKit is unreliable about. A follow-up (`f7745171`) keeps the scrim concentric at the screen edges.

### #766 - the replay control ships
Removes the `NODE_ENV !== 'production'` gate from Settings -> Account -> "Show me around". The gate's own comment named `window.__meridianTour()` as the packaged-build escape hatch, and **that hatch does not exist**: no `devtools` Cargo feature in a release build, and the window is opened as `WebviewUrl::App("today")` with no URL bar, so `?tour=1` is unreachable too. The gate did not make replay developer-only - it made the walkthrough unreachable once taken, on every build a user runs. The copy is corrected too: the old text promised "an example day, not your own", which describes part two only.

### #775 - three walkthrough dead-ends found in a live run
- The tour's narration bar covered the AI-gate modal's first heading. The bar publishes its measured bottom edge as `--mer-tour-say-bottom` and the modal pads itself clear via `max()`, so the handshake is inert when the tour is not running.
- Pressing "Draft with AI" with no provider opens Settings, which takes the shell's single modal slot and unmounts the plan column - so the composer's `useState` died and the user was handed back the board list with their note apparently lost. `isResumePending()` **peeks** rather than consuming, because the flag is `TaskComposer`'s to spend; consuming it in the column would have reopened the composer onto a box it had just emptied.
- Completing the tour re-asked the worklog delivery time, because the tour's copy of the dialog cleared its own latch but not the shell's.

---

### #778 - half the daily summary was inert
Every row under **TODAY'S PLAN** with tracked work behind it - each `Done · 1h 34m` row - was built with `onClick === undefined`, so it rendered a `<div>` rather than a `<button>` and clicking it did nothing. The only plan rows that led anywhere were the **untouched** tickets, which have nothing to file. The draft machinery was never missing: `SummaryTaskView` already had generate / edit / approve / retarget / post, reached from the off-plan rows through the same `onSelect` - the plan block simply had no entry point into it. `drafts` now reaches these rows too, so `DRAFT READY TO POST` shows on a committed ticket and not only on unplanned work.

The judgement call is which strand a row opens when a `PlanVerdict` names several: the **longest**, since that is the strand the row's own displayed duration mostly consists of. Picking the first id would hand the click to whichever strand the adherence join happened to emit first - arbitrary, and silently so.

### #779 - the CodeRabbit review on this PR
Nine findings taken, three declined ([triage](https://github.com/Meridiona/meridian/pull/777#issuecomment-5307861072)). Two were real defects in code this release was about to ship, both in #771's interlock and both unreachable from a macOS run: `refresh.rs` **deleted** the recovery attempt rather than deferring it (the staging gate masked the outputs but still counted the failure, consuming the one-shot `consecutive_health_failures == 2` edge - which matters because `register_service` can report success on Windows while `/Run` and the fallback spawn both failed); and the watchdog's `Start` arm acted on a flag sampled before a socket probe and a forked subprocess, so an install beginning mid-tick restarted the daemon it had just killed.

Also in that PR: XML 1.0 control characters silently killing a Windows toast, a saved wizard position lost exactly when the user granted the permission it asked for, a stale narration-bar measurement on resize, an unchecked cast that could tear down the whole walkthrough, two missing spans, and the What's New backfill described below.

## Everything else

- **#762** - dependabot: bumps the actions group (4 updates).
- **#759** - README rewrite. Its content already reached `main` independently via #763/#765, so this contributes **no README change** to the merge - verified: the merged tree's `README.md` is byte-identical to `main`'s today.

---

## Verification

- `git merge-tree` reports a **clean merge**, and the resulting `README.md` is identical to `main`'s - main's later demo-video and system-requirements edits survive.
- Windows CI passed on every branch that needed it: on #771 (`clippy` 12m58s / `test` 9m46s - the first ever compile of that code), on #779, and here on the merged head (`clippy` 10m40s / `test` 9m26s).
- `cargo test --workspace` 954 pass · `cargo clippy --workspace --all-targets -D warnings` clean · `bun test` 807 pass, run locally against the merged content.

> **One transient failure, for the record.** The `UI` job failed once on this head with `next build` unable to fetch JetBrains Mono - **404s from `fonts.gstatic.com`** cascading into module-not-found. `ui/app/layout.tsx` and `ui/package.json` are untouched by this entire release, the same commit's UI job had passed an hour earlier, and it passed on re-run. Noted because a font-CDN 404 presents as a build error and reads like a code fault.

### What is *not* verified
- **Green CI here does not cover the release build.** The one failure mode worth naming is the `tauri-plugin-notifications` link flake this repo has hit before (`could not find native static library ...`, caused by `rust-cache` restoring `target/` but not the `~/.cargo/registry/src` where the fork's `build.rs` writes its Swift `.a`). `release-build.yml` now caches `registry/src` explicitly, so the cause is addressed rather than worked around - but this window is the first release to add substantial new code against that plugin (#776), so it is the thing to watch if the macOS release build fails. A failure of that shape is not a code defect and a plain re-run will not fix it.
- **#776's interactive toasts have not been exercised on a packaged Windows build** - only compiled and unit-tested.
- **After merge, confirm the back-merge PR is actually created.** That job has silently failed before (`GH_TOKEN` lacking pull-request write, after a clean fast-forward) and left orphaned branches going back to v1.82.1.

### Known gaps, flagged rather than fixed here
- ~~**`whats-new.json` is stale at 1.84.0**~~ - **fixed in #779.** 1.85.0 and 1.86.0 had both shipped without an entry, so anyone updating from 1.84.0 opened a modal two releases short of what they had installed. All three entries (1.85.0, 1.86.0, 1.87.0) are now present.
- **`tray/minimum-version` contains `1.84.0`.** This does not block - the packager only fails when the floor strictly *exceeds* the release version - but it means 1.87.0 will continue to force-update anyone below 1.84.0. Plausibly still wanted after the corruption fixes; noting it so it is a decision rather than a leftover.

### Excluded by cutting now
- **#772** (`feat/tour-story-chrome`) and **#774** (README Product Hunt badge) are still open against `pre-main` and are **not** in this release.
- **#778 and #779 landed on `pre-main` after this PR was opened** and are included - the two sections above describe them. #778 was merged without review; it has since been read line-by-line (`pickPrimary`'s `tasks.indexOf` is safe because `byId` is built from `tasks`, and the index it feeds is cosmetic) and `tsc --noEmit` is clean on app code.
- **#773** targets `main` directly with what looks like the same Product Hunt badge as #774. Worth resolving which one lands, so the badge does not arrive twice.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by Akarsh-Hegde** [comment]
@coderabbitai review

**Comment by Akarsh-Hegde** [comment]
Review triage: **9 findings taken, 3 declined.** Fixes are in #779 against `pre-main` - merging that updates this PR in place.

**Two were real defects in code this release was about to ship**, both in the install/watchdog interlock from #771, and neither reachable from a macOS test run:

1. **`refresh.rs` deleted the recovery attempt rather than deferring it.** The staging gate masked the outputs but still counted the failure, and every action keys off the one-shot `consecutive_health_failures == 2` edge - so a suppressed tick consumed the edge and no later failure could ever recover it. That matters because `register_service` can report success on Windows while `/Run` and the fallback spawn both failed, leaving a permanently down daemon this loop had already spent its one attempt on. Now the count is skipped, so the episode is deferred ~60 s, not lost.
2. **The watchdog's `Start` arm acted on a stale flag.** `decide` is pure and judges a snapshot taken before a socket probe and a forked liveness check - an install beginning in that window started the daemon it had just killed. Re-read at the last instant (not `LIFECYCLE`, for the reason `begin_staging` already documents).

The rest: XML control characters silently killing a Windows toast, a saved wizard position lost exactly when the user granted the permission it asked for, a stale narration-bar measurement on resize, an unchecked cast that could tear down the whole walkthrough, two missing spans, an async fix, and one test whose negative assertion could pass vacuously.

**Declined, with reasons:**
- **Pin every Action to a SHA** (2 findings, 20 sites) - legitimate, but repo-wide workflow work this release does not touch. Making the release ride on 20 unrelated edits to the pipeline that builds it is the wrong trade. Wants its own PR.
- **Remove the 1.86.0 CHANGELOG section** - it is already on `main`; this is the merge-base artifact described in the callout at the top of this PR, not a handwritten section.
- **camelCase `BODY_TOP` / `TOUR_BODY_PAD`** - module consts in `ui/components` are SCREAMING_CASE throughout, so renaming two would reduce consistency, not improve it.

`whats-new.json` is also updated in #779: 1.87.0's entry plus the **1.85.0 and 1.86.0 backfill** flagged above.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,66 @@
...
+* **config:** rebuild .env.example from what the code actually reads ([7a3a331](https://github.com/Meridiona/meridian/commit/7a3a3314417708680f15ec431ec53f81645d0525))
+* **config:** reduce .env.example to the 4 keys the UI cannot provide ([35b139c](https://github.com/Meridiona/meridian/commit/35b139c7d7ddecc7d7b03876d30516ddee78969a))
+* **config:** trim .env.example to the 12 keys with no working default ([d8d289e](https://github.com/Meridiona/meridian/commit/d8d289efb681adac8e11b0a3d3a8f9c98a969ca5))
+* remove dead scripts and move internal docs out of the root ([de95f69](https://github.com/Meridiona/meridian/commit/de95f69ae6753da3607e9d5486671aa5f348653a))
+* **repo:** purge dead references to the removed Python services/ tree ([c34cc5b](https://github.com/Meridiona/meridian/commit/c34cc5bed25cecb348dbfa6d5887775eb6ea02e9))
+* **scripts:** add dev-reset-onboarding for iterating on the setup wizard ([6d16d43](https://github.com/Meridiona/meridian/commit/6d16d43e5196091dbab0bf64373767690703f7db))
+* untrack .mcp.json and drop dead MLX/Python config from .env.example ([2f03a89](https://github.com/Meridiona/meridian/commit/2f03a8916d56a2e650ca34f9f5ed950dc99e3698))
+
```
[the remark this answers is not in the document]
Comment (Akarsh-Hegde) [reply CHANGELOG.md:63]: Not applicable - this section was not hand-written. `git log -1 --format=%an 23f24797` is `semantic-release-bot`, i.e. this IS the generated release commit that semantic-release produced and committed itself. Removing it would delete the bot's own output.

Resolving as a false positive.

File: ui/components/timeline/ModalShell.tsx [hunk 2]
```diff
@@ -9,6 +9,37 @@
...
+ *  below that row, so the first heading was rendered underneath it (the AI-gate
+ *  step's "Do you have a Claude, ChatGPT or Cursor subscription?" vanished
+ *  entirely). Moving the bar was the obvious fix and is the wrong one: both
+ *  modals with un-anchored beats are `scrollInside` and can reach `maxHeight`
+ *  92%, leaving no room above or below to move it TO, and the bar and the panel
+ *  are both centred with overlapping max-widths so they collide by construction.
+ *
+ *  @param base padding this branch uses when the tour is not running. */
+const TOUR_BODY_PAD = (base: string) =>
+  `max(${base}, calc(var(--mer-tour-say-bottom, 0px) + 12px - ${BODY_TOP}px))`
```
[the remark this answers is not in the document]
Comment (Akarsh-Hegde) [reply ui/components/timeline/ModalShell.tsx:41]: Declining: SCREAMING_SNAKE_CASE is this repo's established convention for module-level constants - 42 such declarations across `ui/components/timeline/` and `ui/lib/` (`WEEKDAY_LABELS`, `FALLBACK_PANE_PX`, `GUTTER`, `CARD_GUTTER_GAP`, …). Renaming these two would make the file inconsistent with its neighbours rather than more consistent.

The cited guideline is about ordinary variables; these are compile-time constants, which is the case that convention exists to distinguish.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -80,12 +80,12 @@ jobs:
...
       migrations: ${{ steps.f.outputs.migrations }}
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v7
         with:
           # Nothing in this workflow pushes; don't leave the token in .git/config
           # where any later step or third-party action could read it.
           persist-credentials: false
-      - uses: dorny/paths-filter@v3
+      - uses: dorny/paths-filter@v4
```
[the remark this answers is not in the document]
Comment (Akarsh-Hegde) [reply .github/workflows/ci.yml:88]: Tracking separately rather than in a release-promotion PR. Pinning every action to a commit SHA is a repo-wide workflow change (`ci.yml`, both release workflows) that touches the release path itself - not something to land in the PR that promotes a release to production, where a mistake blocks shipping.

The suggestion is reasonable on its merits; it just needs its own PR and a green release run behind it.

File: .github/workflows/release-prepare.yml [hunk 4]
```diff
@@ -122,7 +122,7 @@ jobs:
       version: ${{ steps.run.outputs.version }}
       tag: ${{ steps.run.outputs.tag }}
     steps:
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@v7
```
[the remark this answers is not in the document]
Comment (Akarsh-Hegde) [reply .github/workflows/release-prepare.yml:125]: Same as the ci.yml thread - action SHA pinning is worth doing, but it modifies the release workflow itself and should not ride along in the promotion PR. Tracking separately.

File: tray/src-tauri/src/commands/system.rs [hunk 5]
```diff
@@ -155,6 +155,21 @@ pub async fn resize_setup_window(
...
+    // that changed `cardHeight` called `set_size` on a maximized window. Win32
+    // has no "resize while staying maximized": SetWindowPos on a WS_MAXIMIZE
+    // window clears the maximized state, so the window visibly snapped back to
+    // its restored size mid-wizard - the user pressed Continue and the window
+    // un-maximized itself. Nothing needs resizing while maximized anyway: the
+    // card's own `transform: scale(...)` in page.tsx already fills whatever
+    // size the window is.
+    if win.is_maximized().unwrap_or(false) {
+        tracing::debug!("setup window is maximized - resize skipped");
+        return Ok(());
```
[the remark this answers is not in the document]
Comment (Akarsh-Hegde) [reply tray/src-tauri/src/commands/system.rs:171]: Already handled at this site. `resize_setup_window` returns early when `win.is_maximized()` is true (the lines immediately above this one), which is exactly the Win32 case raised - `SetWindowPos` on a `WS_MAXIMIZE` window clears the maximized state and the window snaps back to its restored size mid-wizard.

The comment above the guard records why the earlier `is_fullscreen()` check did not catch it: on Windows that tracks the borderless-fullscreen state, not the zoomed one, so it never fired for a maximized window.

File: tray/src-tauri/src/daemon_lifecycle.rs [hunk 6]
```diff
@@ -265,6 +265,76 @@ pub(crate) fn is_paused() -> bool {
...
+/// serialise the two, but the watchdog would then block on it for the length of
+/// an install and start the daemon the instant it was released — correct, but it
+/// makes a 5 s-budgeted [`stop_for_quit`] queue behind a multi-second install,
+/// and a flag keeps [`crate::poll::watchdog::decide`] a pure function that can
+/// be tested for this exact case. The guard is what makes an early `?` return
+/// or a panic mid-install clear the flag rather than wedging the watchdog off
+/// for the rest of the session.
+pub(crate) fn begin_staging() -> StagingGuard {
+    DAEMON_STAGING.store(true, Ordering::Relaxed);
+    StagingGuard
```
[the remark this answers is not in the document]
Comment (Akarsh-Hegde) [reply tray/src-tauri/src/daemon_lifecycle.rs:326]: This is what `begin_staging` / `StagingGuard` already do - `DAEMON_STAGING` is set for the duration of the install and read by `poll::watchdog::decide`, so the watchdog stands down rather than racing the staging transition.

The doc comment on `begin_staging` covers why it is an `AtomicBool` + RAII guard rather than the `LIFECYCLE` mutex: the mutex would serialise them too, but it would make a 5s-budgeted `stop_for_quit` queue behind a multi-second install, and the flag keeps `decide` a pure function testable for exactly this case. The guard is what clears the flag on an early `?` return or a panic mid-install.

File: tray/src-tauri/src/commands/notifications.rs [hunk 7]
```diff
@@ -96,8 +99,31 @@ pub async fn record_notification_response(
...
+    pool: &meridian_core::SqlitePool,
+    id: i64,
+    action: &str,
+    text: Option<&str>,
+) -> Result<(), String> {
     let now = chrono::Utc::now().to_rfc3339_opts(chrono::SecondsFormat::Secs, true);
-    meridian_core::notifications::record_response(pool, id, &action, text.as_deref(), &now)
+    meridian_core::notifications::record_response(pool, id, action, text, &now)
         .await
         .map_err(|e| crate::cmd_err!(e, id, action, "record_notification_response failed"))?;
```
[the remark this answers is not in the document]
Comment (Akarsh-Hegde) [reply tray/src-tauri/src/commands/notifications.rs]: Valid - fixed in #781 (`fix(tray): span the notification delivery and retraction paths`).

Verified the gap was real: `win_toast.rs` and `sys.rs` had zero spans, only bare `tracing::` calls. It matters more than the general rule suggests because every one of these paths crosses a thread - delivery and retraction run inside a `run_on_main_thread` closure (WinRT needs the COM apartment + pump) and the answer write is spawned onto tokio from a WinRT callback, so without a span entered inside each closure the failure and its cause land in telemetry unrelated.

Added `notifications.toast.deliver` (id, category, button count), `notifications.toast.retract` on both arms, and `notifications.toast.answer` which `.instrument`s the spawned write so `apply_response`'s existing `notifications.respond` span becomes its child rather than an orphan root. `otel.status_code` declared and recorded on each.

One correction to the finding: `apply_response` and `record_notification_response` were already instrumented - that landed in #779. This closes the other half.

No user content on any span (no titles, bodies or reply text) - the category and a boolean are the diagnostics, per the redaction allowlist.
